src/sentry/sentry_metrics/multiprocess.py

```diff
@@ -441,25 +447,168 @@ def create(
+        max_batch_time: float,
+        commit_max_batch_size: int,
+        commit_max_batch_time: float,
+    ):
+        self.__max_batch_time = max_batch_time
+        self.__max_batch_size = max_batch_size
+        self.__commit_max_batch_time = commit_max_batch_time
+        self.__commit_max_batch_size = commit_max_batch_size
+
+    def create(
+        self, commit: Callable[[Mapping[Partition, Position]], None]
+    ) -> ProcessingStrategy[KafkaPayload]:
+        last_step = CollectStep(
+            step_factory=SimpleProduceStep,
+            commit_function=commit,
+            max_batch_size=self.__commit_max_batch_size,
+            # convert to seconds
+            max_batch_time=self.__commit_max_batch_time / 1000,
+        )
+        transform_step = TransformStep(next_step=last_step)
+        strategy = BatchMessages(transform_step, self.__max_batch_time, self.__max_batch_size)
+        return strategy
+
+
+class TransformStep(ProcessingStep[MessageBatch]):  # type: ignore
+    """
+    Temporary Transform Step
+    """
+
+    def __init__(
+        self,
+        next_step: ProcessingStep[KafkaPayload],
+    ) -> None:
+        self.__process_messages = process_messages
+        self.__next_step = next_step
+        self.__closed = False
+        self.__metrics = get_metrics()
+
+    def poll(self) -> None:
+        self.__next_step.poll()
+
+    def submit(self, message: Message[MessageBatch]) -> None:
+        assert not self.__closed
+
+        with self.__metrics.timer("transform_step.process_messages"):
+            transformed_message_batch = self.__process_messages(message)
+
+        for transformed_message in transformed_message_batch:
+            self.__next_step.submit(transformed_message)
+
+    def close(self) -> None:
+        self.__closed = True
+
+    def terminate(self) -> None:
+        self.__closed = True
+
+        logger.debug("Terminating %r...", self.__next_step)
+        self.__next_step.terminate()
+
+    def join(self, timeout: Optional[float] = None) -> None:
+        self.__next_step.close()
+        self.__next_step.join(timeout)
+
+
+class UnflushedMessages(Exception):
+    pass
+
+
+class SimpleProduceStep(ProcessingStep[KafkaPayload]):  # type: ignore
+    def __init__(
+        self,
+    ) -> None:
+        snuba_metrics = settings.KAFKA_TOPICS[settings.KAFKA_SNUBA_METRICS]
+        snuba_metrics_producer = Producer(
+            kafka_config.get_kafka_producer_cluster_options(snuba_metrics["cluster"]),
+        )
+        producer = snuba_metrics_producer
+        self.__producer = producer
+        self.__producer_topic = settings.KAFKA_TOPICS[settings.KAFKA_SNUBA_METRICS].get(
+            "topic", "snuba-metrics"
+        )
+
+        self.__closed = False
+        self.__metrics = get_metrics()
+
+    def poll(self) -> None:
+        pass
+
+    def submit(self, message: Message[KafkaPayload]) -> None:
+        with self.__metrics.timer("simple_produce_step.produce_time_duration"):
```

Comment:
Please remove this metric. Now you would be logging it for every message, which would impact the performance itself.